Scalar.Common/Maintenance/ConfigStep.cs

```diff
@@ -66,6 +66,7 @@ private enum GitCoreGVFSFlags
         }
 
         private bool? UseGvfsProtocol = true;
+        private Dictionary<string, GitConfigSetting> existingConfigSettings = null;
```

Comment:
Do we have a known `IEqualityComparer` for this dictionary's keys? Does the method that instantiates this dictionary always return a dictionary with a `StringComparer.OrdinalIgnoreCase` for example?

Reply:
It gets the config values out of `git config`, which does its own key-collapsing based on case.